software/extra_script.py

```diff
@@ -1,17 +1,21 @@
 Import("env")
 
 env.AddCustomTarget(
-    "uploadnobuild",
-    None,
-    'pio run -e %s -t nobuild -t upload' %
-        env["PIOENV"],
-    title="Uploads without building"
+    name="uploadnobuild",
+    dependencies=None,
+    actions=[
+        "pio run -e $PIOENV -t nobuild -t upload",
+    ],
+    title="Uploads without building via serial"
 )
 
 env.AddCustomTarget(
     name="ota",
     dependencies="$BUILD_DIR/${PROGNAME}.elf",
-    actions=["$PROJECT_DIR/ota-flasher.sh $PROJECT_DIR $PIOENV $PROGRAM_ARGS"],
+    actions=[
+        "pio run -e $PIOENV",
```

Comment:
is this line the trick?

Reply:
yes, that calls pio run for a normal build before the ota-flasher.sh script